Signal/src/view controllers/SignalsViewController.m

```diff
@@ -138,6 +138,44 @@ - (void)viewDidLoad {
                                              selector:@selector(handleDismissCallInterstitialNotification:)
                                                  name:[CallService dismissCallInterstitialNotificationName]
                                                object:nil];
+    
+    [self updateBarButtonItems];
+}
+
+- (void)updateBarButtonItems {
+    const CGFloat kBarButtonSize = 44;
+    if (YES) {
+        // We use UIButtons with [UIBarButtonItem initWithCustomView:...] instead of
+        // UIBarButtonItem in order to ensure that these buttons are spaced tightly.
+        // The contents of the navigation bar are cramped in this view.
+        UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
+        UIImage *image = [UIImage imageNamed:@"button_settings_white"];
+        [button setImage:image
+                forState:UIControlStateNormal];
+        UIEdgeInsets imageEdgeInsets = UIEdgeInsetsZero;
+        // We normally would want to use left and right insets that ensure the button
+        // is square and the icon is centered.  However UINavigationBar doesn't offer us
+        // control over the margins and spacing of its content, and the buttons end up
+        // too far apart and too far from the edge of the screen. So we use a smaller
+        // left inset tighten up the layout.
+        imageEdgeInsets.right = round((kBarButtonSize - image.size.width) * 0.5f);
```

Comment:
Yeah... so, my rule of thumb is to DRY stuff up if its bulky OR if you do it three times or more.  These look similar, but the "left hand" button logic is different than the "right hand" logic.  I am tempted.  Maybe I'll do it this weekend.